deploy/docker-setup.sh

```
@@ -1,42 +1,51 @@
-#!/bin/bash
+#!/usr/bin/bash
 set -e
 
-# MariaDB needs a second
-sleep 5
+GEN_PWD="$(openssl rand -base64 42)"
+EDUMFA_ADMIN_USER="${EDUMFA_ADMIN_USER:-admin}"
+EDUMFA_ADMIN_PASS="${EDUMFA_ADMIN_PASS:-$GEN_PWD}"
 
 # Create enckey if doesn't exist yet
-if [ ! -f /etc/edumfa/enckey ]; then
-  edumfa-manage create_enckey
-fi
+edumfa-manage -q create_enckey || true
 
 # Create audit keys if they don't exist yet
-if [[ ! -f /etc/edumfa/pubkey.pem && ! -f /etc/edumfa/private.pem ]]; then
-  edumfa-manage create_audit_keys
-fi
+edumfa-manage -q create_audit_keys || true
 
 # Create DB
 echo "Creating DB"
-edumfa-manage createdb
+# FIXME: this creates a exception trace on every attempt
+until edumfa-manage -q create_tables
+do
+    echo "Cannot connect to database. Trying again..."
+    sleep 3
+done
```

Shouldn't this be resolved before merging? Also do we maybe want a timeout?

---

> Also do we maybe want a timeout?

Do you mean stop after x tries and exit unsuccessfully?

---

> Shouldn't this be resolved before merging?

It would be nice to eventually fix this, but is it a showstopper? I am not sure how to quickly implement this correctly. The "naive" way would be to catch sqlalchemy.exc.OperationalError from `db.create_all()`,  but that exception class sounds too broad. I would like to avoid hiding important errors from the user.

---

> Do you mean stop after x tries and exit unsuccessfully?

I think it would be better practice. At some point we can shutdown the container.

> It would be nice to eventually fix this, but is it a showstopper? I am not sure how to quickly implement this correctly. The "naive" way would be to catch sqlalchemy.exc.OperationalError from `db.create_all()`,  but that exception class sounds too broad. I would like to avoid hiding important errors from the user.

Aight get it. Just needed the reasoning behind it. I get what you mean. Not sure if we even want to silence this error, as it shouldn't be raised if everything was done right?

---

> I think it would be better practice. At some point we can shutdown the container.

Done, it now exits after 10 tries.

> Aight get it. Just needed the reasoning behind it. I get what you mean.

I am sorry if I overexplained!

> Not sure if we even want to silence this error, as it shouldn't be raised if everything was done right?

It will get raised if the container starts and the DB is not available yet. For example if the database is part of the same compose file, it might appear once or twice. In the example compose file in #781 this should not happen since we have a health check for the database.

---

Sounds good